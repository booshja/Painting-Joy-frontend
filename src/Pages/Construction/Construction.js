// dependencies
import React from "react";
import styled from "styled-components";
// images
import Image from "../../images/paint-bucket.png";

const ConstructionDiv = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 1rem;
`;

const ConstructionHeader = styled.h1`
    font-weight: 700;
    text-align: center;
    font-size: 1.8rem;
    padding: 2rem 0;
`;

const ConstructionImg = styled.img`
    width: 200px;
    margin-bottom: 2rem;
`;

const ConstructionP = styled.p`
    text-align: center;
    line-height: 1.5;
`;

const ConstructionA = styled.a`
    font-weight: 700;
    text-decoration: underline;
    color: rgb(0, 0, 0);

    &:hover {
        color: rgb(144, 85, 162);
    }
`;

const CopyrightP = styled.p`
    margin-top: 2rem;
    font-size: 0.8rem;
`;

function Construction() {
    return (
        <ConstructionDiv>
            <ConstructionHeader>Masterpiece In Progress...</ConstructionHeader>
            <ConstructionImg src={Image} alt="Paintbrush and paint can" />
            <ConstructionP>
                In the meantime, find me on {/* */}
                <ConstructionA href="https://www.instagram.com/paintingjoymuralco/">
                    Instagram,
                </ConstructionA>
            </ConstructionP>
            <ConstructionP>
                Or shoot me an email at: {/* */}
                <ConstructionA href="mailto:paintingjoymuralco@gmail.com">
                    paintingjoymuralco@gmail.com
                </ConstructionA>
            </ConstructionP>
            <CopyrightP>&#169; 2021 - Painting Joy Mural Co.</CopyrightP>
        </ConstructionDiv>
    );
}

export default Construction;
