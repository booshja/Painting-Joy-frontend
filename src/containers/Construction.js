// dependencies
import React from "react";
import styled from "styled-components";
// breakpoints
import { breakpoints } from "../breakpoints";
// assets
import Logo from "../assets/painting-joy-logo-white.jpg";

const StyledConstruction = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    height: 100%;
    min-height: 97vh;
`;

const StyledContent = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`;

const StyledImg = styled.img`
    width: 90%;

    ${breakpoints("width", "px", [{ 1024: 400 }])}
`;

const StyledHeadline = styled.h1`
    font-family: "Barlow Condensed";
    font-size: 3rem;
    letter-spacing: 1px;
`;

const StyledText = styled.p`
    font-family: "News Cycle", sans-serif;
    font-size: 1.4rem;
    text-align: center;
    letter-spacing: 1px;

    ${breakpoints("font-size", "rem", [{ 414: 1.6 }])}
`;

const StyledLink = styled.a`
    font-size: 3rem;
    color: #000000;

    ${breakpoints("font-size", "rem", [{ 414: 4 }])}

    &:hover {
        color: #207070;
        transform: translateY(-1px);
    }
`;

const StyledCopyright = styled.p`
    font-size: 1rem;
    justify-self: flex-end;
    text-align: center;
`;

const Construction = () => {
    return (
        <StyledConstruction>
            <StyledContent>
                <StyledHeadline>Coming Soon!</StyledHeadline>
                <StyledImg src={Logo} />
                <StyledText>
                    In the meantime, check out my Instagram!
                </StyledText>
                <StyledLink href="https://www.instagram.com/paintingjoymc/">
                    <i className="fab fa-instagram"></i>
                </StyledLink>
            </StyledContent>
            <StyledCopyright>
                &copy; - 2021 Painting Joy Mural Co.
            </StyledCopyright>
        </StyledConstruction>
    );
};

export default Construction;
