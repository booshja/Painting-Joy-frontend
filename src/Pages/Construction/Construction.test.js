import React from "react";
import { render } from "@testing-library/react";
import Construction from "./Construction";

// Smoke Test
it("renders without breaking", () => {
    render(<Construction />);
});

// Snapshot Test
it("matches snapshot", () => {
    const { asFragment } = render(<Construction />);
    expect(asFragment()).toMatchSnapshot();
});

// General testing
it("displays the image", () => {
    const { queryByAltText } = render(<Construction />);

    // expect the image to show
    expect(queryByAltText("Paintbrush and paint can")).toBeInTheDocument();
});
