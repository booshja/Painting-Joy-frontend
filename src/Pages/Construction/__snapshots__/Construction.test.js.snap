// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM bfNuOb"
  >
    <h1
      class="sc-gtsrHT jXymxE"
    >
      Masterpiece In Progress...
    </h1>
    <img
      alt="Paintbrush and paint can"
      class="sc-dlnjwi jfTtog"
      src="paint-bucket.png"
    />
    <p
      class="sc-hKFxyN ktZtmE"
    >
      In the meantime, find me on 
      <a
        class="sc-eCApnc hliTlW"
        href="https://www.instagram.com/paintingjoymuralco/"
      >
        Instagram,
      </a>
    </p>
    <p
      class="sc-hKFxyN ktZtmE"
    >
      Or shoot me an email at: 
      <a
        class="sc-eCApnc hliTlW"
        href="mailto:paintingjoymuralco@gmail.com"
      >
        paintingjoymuralco@gmail.com
      </a>
    </p>
    <p
      class="sc-jSFjdj eLTBFC"
    >
      Â© 2021 - Painting Joy Mural Co.
    </p>
  </div>
</DocumentFragment>
`;
